tools/dev_setup.sh

```diff
@@ -19,10 +19,37 @@ EOD
 
 vagrant up
 
+# fix permissions
+cat <<EOD | vagrant ssh
+sudo chown vagrant /home/vagrant/go
+sudo chown vagrant /home/vagrant/go/src/
+sudo chown vagrant /home/vagrant/go/src/github.com
+EOD
+
+# golang setup
+cat <<EOD | vagrant ssh
+mkdir -p /opt
+wget -q https://dl.google.com/go/go1.11.2.linux-amd64.tar.gz
+sudo tar -xzf go1.11.2.linux-amd64.tar.gz -C /opt
+echo 'export GOPATH=/home/vagrant/go' > ~/.bashrc
+echo 'export PATH=/opt/go/bin:/home/vagrant/go/bin:\$PATH' >> ~/.bashrc
+
+sudo wget https://github.com/golang/dep/releases/download/v0.5.0/dep-linux-386 -O /usr/local/bin/dep -q
```

Comment:
Is there no way to get the versions we want through the normal package manager?

Reply:
I would have preferred that too, but unfortunately not inside ubuntu. The recommendations are:

- use brew on osx
- use a `curl http://...install.sh | bash` script which is a little scary
- or use one of the precompiled releases
- `go get` which uses the version found on master, which can sometimes subtly break the latest tagged version

https://github.com/golang/dep#installation

For go and protobuf, the image the VM is based on uses apt repositories which have pretty old releases. Docs for both go and protobuf suggest installing from a tarballs, so I opted to do that instead of finding newer repositories to download from.